quart_prometheus_logger.py

```diff
@@ -127,46 +140,57 @@ def _register_collectors(self):
                     "http_requests",
                     "Total number of requests",
                     ["method", "path", "status", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Counter(
                     "http_requests_errors",
                     "Total number of error requests",
                     ["method", "path", "status", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_request_duration_seconds",
                     "The amount of time spent handling requests",
                     ["path", *self._custom_label_names],
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_request_size_bytes",
                     "The size of requests",
                     ["path", *self._custom_label_names],
                     buckets=REQUEST_BUCKETS,
+                    registry=self._registry,
                 ),
                 Histogram(
                     "http_response_size_bytes",
                     "The size of responses",
                     ["path", *self._custom_label_names],
                     buckets=RESPONSE_BUCKETS,
+                    registry=self._registry,
                 ),
             )
         }
 
+    def _reset_metrics(self):
+        """Unregister all collectors from the registry."""
+        for _, collector in self._collectors.items():
+            self._registry.unregister(collector)
+
     def custom_route_labeler(
         self, labeler: Callable[["LocalProxy"], Dict[str, str]], label_names: List[str]
     ) -> None:
         """Add a handler for providing additional labels for a route.
 
         This will reset all metrics. Conventionally it's called when the extension is first registered
 
-        :param labeler: The handler function to invoke. It must return a dict of key-value labels.
+        :param labeler: The handler function to invoke. It takes a Quart LocalProxy representing
+                    the request, from which values for the custom label can be pulled. It must return
```

Comment:
Instead of "pulled", maybe something a little more active like "constructed" or "generated"?